src/app/cli/src/explore/api_server.rs

```diff
@@ -63,15 +69,21 @@ impl APIServer {
 
         let server = axum::Server::bind(&addr).serve(app.into_make_service());
 
-        Self { server }
+        Self {
+            server,
+            task_executor,
+        }
     }
 
     pub fn local_addr(&self) -> SocketAddr {
         self.server.local_addr()
     }
 
-    pub async fn run(self) -> Result<(), hyper::Error> {
-        self.server.await
+    pub async fn run(self) -> Result<(), InternalError> {
+        tokio::select! {
+            res = self.server => { res.int_err() },
+            res = self.task_executor.run() => { res.int_err() },
+        }
```

Comment:
Notice: `kamu system api-server` now spawns an HTTP server and a task executor loop.